MAINT: improve error with SMILES

tylerjereddy opened the pull request:
* The error message when processing conventional SMILES strings can be confusing, as observed at:
https://github.com/marrink-lab/polyply_1.0/issues/390#issuecomment-2455472121

* For cases like that where there are no polymeric fragments (i.e., no use of braces), this patch improves the error message (and adds a test for it) to nudge the user to BigSMILES or CGSmiles formats.

fgrunewald commented on cgsmiles/resolve.py:
```diff
@@ -430,6 +430,9 @@ def from_string(cls, cgsmiles_str, last_all_atom=True, legacy=False):
         elements = re.findall(r"\{[^\}]+\}", cgsmiles_str)
+        if len(elements) == 0:
+            raise SyntaxError("No polymeric fragments detected; likely "
+                              "not a valid CGSmiles/BigSMILES string.")
```
at the moment it only parses CGSmiles; can you remove the BigSMILES?

fgrunewald commented on cgsmiles/tests/test_molecule_resolve.py:
```diff
@@ -310,6 +310,8 @@ def _atomname_match(n1, n2):
  ("{[#A][#B]1}.{#A=CC[$],#B=OC[$]}", "You have a dangling ring index."),
  ("{[#A]1[#B]1}{#A=CC[$],#B=OC[$]}", "You define two edges between the same node. Use bond order symbols instead."),
+ # see: https://github.com/marrink-lab/polyply_1.0/issues/390#issuecomment-2455472121
+ ("CC(c1ccccc1)CC(c1ccccc1)CCCCCCCCCC(CC)CC(CC)CC(c1ccccc1)CC(c1ccccc1)", "No polymeric fragments detected; likely not a valid CGSmiles/BigSMILES string."),
```
don't forget to update the error message here as well

fgrunewald reviewed (changes requested):
Thanks for your contribution. Since we only parse CGSmiles at the moment, I'd remove BigSMILES from the error message.

tylerjereddy commented:
I pushed a commit that attempts to resolve the first round of review comments.

Also, I noticed that `test_syntax_errors` wasn't actually verifying that the error message strings matched, so I went ahead and fixed that by using the canonical `match=..` syntax, and that now correctly enforces match error strings (I fixed one small issue that slipped through previously because of this).
